core/trino-spi/src/main/java/io/trino/spi/eventlistener/ColumnDetail.java

```diff
@@ -13,19 +13,22 @@
  */
 package io.trino.spi.eventlistener;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.Objects;
 
 import static java.util.Objects.requireNonNull;
 
+// JSON serialization is not guaranteed to be stable between releases
```

Comment:
The comment like this is unlikely to get noticed.

Isn't this _the_ reason why we didn't have proper JSON serialization here?
and we intentionally didn't add JSON serialization to `ConnectorMaterializedViewDefinition`, because we didn't want to maintain backwards compatibility?

cc @martint @sopel39

Reply:
My experience is that connector should serialize on it's own. This allows to change SPI without risk of breaking compatibility of serialized data, which is much harder that just changing signature of SPI classes.
Relevant examples are `ConnectorViewDefinition`, which is much harder to change because of all potential serializations. For this purpose `ConnectorMaterializedViewDefinition` is not serializable
It seems that event listener classes might change even more.

Reply:
From what seems reasonable, it seems like either **all** of the pojo classes in the event-listener SPI should be jackson serializable (with a warning that this is not guaranteed to be consistent), or **none** of them should be. Right now maybe 1/3 of them are jackson serializable which is a much more confusing situation.